pkg/ans/ans_test.go

```diff
@@ -0,0 +1,236 @@
+				eventBody, _ := ioutil.ReadAll(req.Body)
+				event := &Event{}
+				json.Unmarshal(eventBody, event)
+				assert.Equal(t, eventDefault, *event, "Mismatch in requested event body")
+			})
+		})
+		t.Run("on status 202", func(t *testing.T) {
+			err := examinee.doSend(eventDefault, func(rw http.ResponseWriter, req *http.Request) {
+				rw.WriteHeader(http.StatusAccepted)
+			})
+			require.NoError(t, err, "No error expected.")
+		})
+	})
+
+	t.Run("bad", func(t *testing.T) {
+		t.Run("on status 400", func(t *testing.T) {
+			err := examinee.doSend(eventDefault, func(rw http.ResponseWriter, req *http.Request) {
+				rw.WriteHeader(http.StatusBadRequest)
+				rw.Write([]byte("an error occurred"))
+			})
+			assert.Error(t, err)
+			assert.Contains(t, err.Error(), "Did not get expected status code 202")
+		})
+	})
+}
+
+func TestANS_CheckCorrectSetup(t *testing.T) {
+	examinee := Examinee{}
+	defer examinee.finish()
+	examinee.init()
+
+	t.Run("good", func(t *testing.T) {
+		t.Run("pass request attributes", func(t *testing.T) {
+			examinee.doCheckCorrectSetup(func(rw http.ResponseWriter, req *http.Request) {
+				assert.Equal(t, http.MethodGet, req.Method, "Mismatch in requested method")
+				assert.Equal(t, "/cf/consumer/v1/matched-events", req.URL.Path, "Mismatch in requested path")
+				assert.Equal(t, "bearer 1234", req.Header.Get(authHeaderKey), "Mismatch in requested auth header")
+				assert.Equal(t, "application/json", req.Header.Get("Content-Type"), "Mismatch in requested content type header")
+			})
+		})
+		t.Run("on status 200", func(t *testing.T) {
+			err := examinee.doCheckCorrectSetup(func(rw http.ResponseWriter, req *http.Request) {
+				rw.WriteHeader(http.StatusOK)
+			})
+			require.NoError(t, err, "No error expected.")
+		})
+	})
+
+	t.Run("bad", func(t *testing.T) {
+		t.Run("on status 400", func(t *testing.T) {
+			err := examinee.doCheckCorrectSetup(func(rw http.ResponseWriter, req *http.Request) {
+				rw.WriteHeader(http.StatusBadRequest)
+				rw.Write([]byte("an error occurred"))
+			})
+			assert.Error(t, err)
+			assert.Contains(t, err.Error(), "Did not get expected status code 200")
+		})
+	})
+}
+
+func TestUnmarshallServiceKey(t *testing.T) {
+	tests := []struct {
+		name              string
+		serviceKeyJSON    string
+		wantAnsServiceKey ServiceKey
+		wantErr           bool
+	}{
+		{
+			name: "Proper event JSON yields correct event",
+			serviceKeyJSON: `{
+						"url": "https://my.test.backend",
+						"client_id": "myTestClientID",
+						"client_secret": "super secret",
+						"oauth_url": "https://my.test.oauth.provider"
+					}`,
+			wantAnsServiceKey: ServiceKey{
+				Url:          "https://my.test.backend",
+				ClientId:     "myTestClientID",
+				ClientSecret: "super secret",
+				OauthUrl:     "https://my.test.oauth.provider",
+			},
+			wantErr: false,
+		},
+		{
+			name:           "Faulty JSON yields error",
+			serviceKeyJSON: `bli-da-blup`,
```

Comment:
bli-da-blup!?

Reply:
I guess it's unprofessional. Just a bit of lonesome home office fun. I'll change it throughout.